osbuild/buildroot.py

```diff
@@ -263,3 +274,27 @@ def run(self, argv, monitor, binds=None, readonly_binds=None):
         data.close()
 
         return CompletedBuild(proc, output)
+
+    @classmethod
+    def read_with_timeout(cls, proc, start, timeout=600000):
+        READ_ONLY = select.POLLIN | select.POLLPRI | select.POLLHUP | select.POLLERR
+        buf = None
+        remaining = timeout - (time.monotonic() - start)
+        if remaining <= 0:
+            proc.terminate()
+            raise StageTimeoutError()
+
+        fd = proc.stdout.fileno()
+        poller = select.poll()
+        poller.register(fd, READ_ONLY)
+        events = poller.poll(remaining)
```

Comment:
I wonder how big the overhead is, creating the poll all the time, adding a bunch of extra syscalls. Probably not a lot compared to all the I/o osbuild is doing.

Reply:
When testing it is still fast, however, Ill move the register and unregister outside this function :smile: 

Try and make it as efficient as possible